selenium_tests/class_roster/class_roster_permission_tests.py

```diff
@@ -0,0 +1,43 @@
+from ddt import ddt, data, unpack
+
+from selenium_common.base_test_case import get_xl_data
+from selenium_tests.class_roster.class_roster_base_test_case import \
+    CLASS_ROSTER_PERMISSION_ROLES
+from selenium_tests.course_admin.course_admin_base_test_case import \
+    CourseAdminBaseTestCase
+
+
+@ddt
+class ClassRosterPermissionTests(CourseAdminBaseTestCase):
+
+    @data(*get_xl_data(CLASS_ROSTER_PERMISSION_ROLES))
+    @unpack
+    def test_roles_access(self, user_id, given_access):
+        # This test masquerades as users in roles in the spreadsheet
+        # specified in CLASS_ROSTER_PERMISSION_ROLES, and then validates the
+        # the users are granted/denied access based on their role.
+
+        self.masquerade_page.masquerade_as(user_id)
+        # Go back to the Manage Course Dashboard
+        self.driver.get(self.TOOL_URL)
+
+        if given_access == 'yes':
+            self.assertTrue(
+                self.course_admin_dashboard_page
+                    .class_roster_button_is_displayed(),
+                'User {} should have access to see tool but does not'.format(
+                    user_id)
+            )
```

Comment:
@hktest If they _do_ see the button but when they click on it they are denied (they see an error), that would be a bug. We may want to check for that, since it's just another line or two of code.